### DISCUSSION TITLE
Can I hide the default webhook endpoint from swagger?

### DISCUSSION THREAD

**Discussion by nkusibojoski** [golevelup/nestjs#549] [Q&A]
In the documentation, I found this: "This module will automatically add a new API endpoint to your NestJS application for processing webhooks."

Is there a way to remove/hide this endpoint?
I am using different way/endpoint for handling routes and also I don't want this endpoint to be exposed in the swagger.

Thanks :)

**Comment by WonderPanda** [comment]
Are you registering the service that actually processes the webhooks separately and then somehow setting up a different route to it?

I'm not sure what you mean by removing it. Without it, the package doesn't really have a purpose as its main goal is to help you automatically route and process incoming webhooks from Stripe. 

If you just want to have an injectable Stripe client it would be pretty trivial to add that to your app without needing ot use this package at all

**Comment by nkusibojoski** [comment]
I was using different package and now switched to this one. Previously I created the route manually and it is quite different than this one. So, at this moment, you are right I am using this package just to have an Injectable Stripe Client and that is why I want to hide/remove this default route from swagger and at all. 

What do you suggest is the best?

**Reply by WonderPanda** [reply]
If you're not going to use the Webhook functionality, I'd just suggest that you provide the Stripe client injectable yourself. The point of this library is to make handling webhooks easy. Feel free to look at the source and just grab the part that provides the Stripe client and copy and paste it into your project

**Reply by laserchicken** [reply]
I'm not sure if I understood the problem correctly either, but I think you can add `ApiExcludeController()` to the controller decorators (see https://github.com/golevelup/nestjs/tree/master/packages/stripe#webhook-controller-decorators) to exclude it from swagger. @nkusibojoski
